gsa/src/web/pages/reports/alertactions.js

```diff
@@ -81,49 +92,95 @@ class AlertActions extends React.Component {
     }
   }
 
-  handleTestAlert() {
-    const {alert_id} = this.state;
+  handleTriggerAlert(state) {
     const {
+      alertId,
+      applyOverrides,
+      includeNotes,
+      includeOverrides,
+      storeAsDefault,
+    } = state;
+    const {
+      filter,
       gmp,
       report,
-      filter,
       showErrorMessage,
       showSuccessMessage,
     } = this.props;
 
+    filter.set('notes', includeNotes);
+    filter.set('overrides', includeOverrides);
+    filter.set('apply_overrides', applyOverrides);
+
     this.handleInteraction();
 
+    if (storeAsDefault) {
+      const defaults = {
+        applyOverrides,
+        includeNotes,
+        includeOverrides,
+      };
+      this.props.saveReportComposerDefaults(defaults);
+    }
+
     gmp.report.alert({
       report_id: report.id,
-      alert_id,
+      alert_id: alertId,
       filter: filter.simple(),
     }).then(response => {
-      showSuccessMessage(_('Running the alert was successfull'));
+      showSuccessMessage(_('Running the alert was successful'));
+      this.setState({showTriggerAlertDialog: false});
     }, error => {
         log.error('Failed running alert', error);
         showErrorMessage(_('Failed to run alert.'));
+        this.setState({showTriggerAlertDialog: false});
       }
     );
   }
 
-  onAlertCreated(response) {
-    const {alerts} = this.state;
-    const alert = response.data;
+  handleOpenTriggerAlertDialog() {
+    this.setState({
+      ...this.props.reportComposerDefaults,
+      showTriggerAlertDialog: true,
+    });
+  }
 
+  handleCloseTriggerAlertDialog() {
     this.setState({
-      alerts: [alert, ...alerts],
-      alert_id: alert.id,
+      showTriggerAlertDialog: false,
     });
   }
 
+  onAlertCreated(response) {
+    const {alerts} = this.state;
+    const {gmp} = this.props;
+    gmp.alert.get({id: response.data.id}).then(resp => {
```

Comment:
Would also be better to load the alert from the store too but that's a bit more difficult.